src/Request/Request.php

```diff
@@ -421,21 +421,34 @@ public function request($path, array $options = []): RequestOperationResult
             $options
         );
         $body = $response->getBody()->getContents();
-        try {
-            $body = \json_decode(
-                $body,
-                false,
-                512,
-                JSON_THROW_ON_ERROR
-            );
-        } catch (\JsonException $jsonException) {
-            $this->logger->debug($jsonException->getMessage());
+        $statusCode = $response->getStatusCode();
+        $headers = $response->getHeaders();
+
+        // Don't attempt to decode JSON if the response is expected to have no body.
+        if (
+            !in_array($statusCode, [204, 304], true)
```

Comment:
It would be helpful to be more explicit (i.e., explain in a comment) the significance of these particular status codes, and why this check is important to do in addition to the content-length / empty body checks.

Reply:
I moved this logic around a bit to try to make it clearer